src/compiler/crystal/compiler.cr

```diff
@@ -295,7 +297,15 @@ module Crystal
 
       Crystal.timing("Codegen (linking)", @stats) do
         Dir.cd(output_dir) do
-          system %(#{CC} -o "#{output_filename}" "${@}" #{@link_flags} #{lib_flags}), object_names
+          if thin_lto
+            lto_cache_dir = "#{output_dir}/lto.cache"
+            Dir.mkdir_p(lto_cache_dir)
+            cc = ENV["CC"]? || "clang -flto=thin -Wl,-mllvm,-threads=#{n_threads},-cache_path_lto,#{lto_cache_dir},#{@release ? "-mllvm,-O2" : "-mllvm,-O0"}"
```

Comment:
embedding lld might be a good way to go in the long term. I wanted to change as little as possible.

@ysbaddaden yes, I wanted to be able to override the linker as you can do using CC. Since the flags are sensible to the linker to be used it basically boils down to the user to set the rest of the flags as well.